datalad/local/subdatasets.py

```diff
@@ -284,21 +281,33 @@ def _get_submodules(ds, paths, fulfilled, recursive, recursion_limit,
     repo = ds.repo
     if not GitRepo.is_valid_repo(dspath):
         return
+    # expand all test cases for the contains test in the loop below
+    # leads to ~20% speedup per loop iteration of a non-match
+    expanded_contains = [
+        # convert to strings, as sm_path below will be a string and
+        # we do not want repeated type-conversion
```

Comment:
This comment isn't accurate.  `c` and `c.parents` aren't converted, and `sm_path` is a path object.

Reply:
When I was pushing fixups for the tests, I removed this comment.  If I'm mistaken, of course feel free to restore it.

Reply:
All good, thx!